components/ugfx/board_framebuffer.h

```diff
@@ -44,30 +42,19 @@ uint8_t target_lut;
 		g->g.Backlight = 100;
 		g->g.Contrast = 50;
 		fbi->linelen = g->g.Width;
-		fbi->pixels = framebuffer;
+		fbi->pixels = ugfx_framebuffer;
 		target_lut = 3;
 	}
 
 	#if GDISP_HARDWARE_FLUSH
 		static void board_flush(GDisplay *g) {
 			(void) g;
 //			ets_printf("Flushing framebuffer!\n");
-//			ets_printf("First byte: %d\n", framebuffer[0]);
-			for (int i = 0; i < BADGE_EINK_WIDTH * BADGE_EINK_HEIGHT / 8; i++) {
-				target_buffer[i] =
-					framebuffer[i * 8 + 7] << 7 |
-					framebuffer[i * 8 + 6] << 6 |
-					framebuffer[i * 8 + 5] << 5 |
-					framebuffer[i * 8 + 4] << 4 |
-					framebuffer[i * 8 + 3] << 3 |
-					framebuffer[i * 8 + 2] << 2 |
-					framebuffer[i * 8 + 1] << 1 |
-					framebuffer[i * 8];
-			}
+//			ets_printf("First byte: %d\n", ugfx_framebuffer[0]);
 			if (target_lut == 0)
-				badge_eink_display(target_buffer, 0);
+				badge_eink_display_one_layer(ugfx_framebuffer, DISPLAY_FLAG_GREYSCALE);

```

Comment:
Let's keep things simple. We also have to think of how we pass on the grey-scale flag.

(btw, in the current API DISPLAY_FLAG_GREYSCALE is a bit poorly chosen. It is both used for '8 bit buffer' and 'draw <n> bits of greyscale')